### PR TITLE
Preliminary style guide for Stylus

### PR DESCRIPTION
**Pull request by jgorset**

### CODE REVIEW

**Comment by jgorset** [comment]
I documented the conventions found in the majority of existing projects. See anything you like, @hyperoslo/web? Or anything you don't?

**Comment by timkurvers** [comment]
👍  We should strongly consider something like [PostCSS modules](https://github.com/css-modules/postcss-modules), not sure if this is the place to mention that 😄

File: CODE_STYLE.md [hunk 1]
```diff
@@ -2,3 +2,14 @@
...
 * [Ruby](https://github.com/bbatsov/ruby-style-guide)
 * [Objective-C](https://github.com/hyperoslo/objective-c-style-guide)
+* [JavaScript](https://github.com/hyperoslo/eslint-config)
+
+## Stylus
+
+We have not yet chosen a style guide for Stylus, other than to decide that:
+
+* We *don't* use `;`.
+* We *do* use `{` and `}`.
```
Comment (mathiasbno) [inline CODE_STYLE.md]: I dont really like `{}`, but I think there is a common consensus to do so.

Comment (jsirevag) [inline CODE_STYLE.md]: I'm with you though..

Comment (gauravtiwari) [inline CODE_STYLE.md]: Yeah me to @mathiasbno - in age of stylus and sass

Comment (jgorset) [inline CODE_STYLE.md]: Me too, I would prefer to not use `{}` like we don't use `;`. There's just no need in my mind.

Comment (mathiasbno) [inline CODE_STYLE.md]: Ahh, oki then xD

Thought I was alone in this world.

Comment (jgorset) [inline CODE_STYLE.md]: @hyperoslo/web, vote!

👍 means you _don't_ want to use `{` and `}`.
👎 means you _do_ want to use `{` and `}`.

**Comment by gauravtiwari** [comment]
As @mathiasbno pointed out, perhaps we should update the doc for SASS and Stylus too? Guess that's what we use mostly nowadays.

File: CODE_STYLE.md (hunk 1 above)
Comment (timkurvers) [inline CODE_STYLE.md]: :+1: on landing something that fits everybody here. The reasoning behind rolling with `{`s on some projects has been to lower mental overhead when swapping between strict ES6 and #do-whatever languages like Stylus. Personally fine with whatever we end up with 👍

**Comment by jgorset** [comment]
I think that'd be cool, too, @timkurvers. It doesn't go against these rules, though, so I think it'd be cool to get this merged as a first step. Feel free to make an issue to discuss and vote on using PostCSS and PostCSS modules as a convention!

**Comment by jgorset** [comment]
Not sure I understand, @gauravtiwari; this is indeed Stylus, and isn't that what you mean?

File: CODE_STYLE.md (hunk 1 above)
Comment (jgorset) [inline CODE_STYLE.md]: Cool. It seems the consensus is to drop them, so I'll update the rule accordingly.

**Comment by mathiasbno** [comment]
Can't we just change the rules to say `We *don't* use `{` and `}`.`?

**Comment by jgorset** [comment]
I did, @mathiasbno.

**Comment by gauravtiwari** [comment]
I mean what @mathiasbno said @jgorset

**Comment by jgorset** [comment]
When, @gauravtiwari?

**Comment by gauravtiwari** [comment]
2 comments above your last comment 😛

**Comment by jgorset** [comment]
Haha, okay. I'm confused, but I think we're agreed now, right?

**Comment by gauravtiwari** [comment]
Yeah, totally.

File: CODE_STYLE.md (hunk 1 above)
Comment (sindrenm) [reply CODE_STYLE.md]: Yeah, I'm also generally for the curly braces as I find it it makes nested selectors more readable, but I'm mostly for finding a standard we can all live with, so :+1: for removing them. :wink:

**Pull request merged**